traits/tests/test_list_events.py

```diff
@@ -192,3 +192,39 @@ def test_clear_empty_list(self):
         foo.l.clear()
 
         self.assertEqual(len(foo.l_events), 0)
+
+    def test_delete_step_slice(self):
+        foo = MyClass()
+        foo.l = [0, 1, 2, 3, 4]
+
+        del foo.l[0:5:2]
+
+        self.assertEqual(len(foo.l_events), 1)
+        event, = foo.l_events
+        self.assertEqual(event.index, slice(0, 5, 2))
+        self.assertEqual(event.removed, [0, 2, 4])
+        self.assertEqual(event.added, [])
+
+    def test_delete_step_slice_empty_list(self):
+        foo = MyClass()
+        foo.l = []
+
+        del foo.l[::-1]
+
+        self.assertEqual(len(foo.l_events), 1)
+        event, = foo.l_events
+        self.assertEqual(event.index, slice(None, None, -1))
+        self.assertEqual(event.removed, [])
+        self.assertEqual(event.added, [])
```

Comment:
The behaviour of this test would be changed by #740 as no list event would be fired